src/catch2/internal/catch_istream.cpp

```diff
@@ -88,27 +97,52 @@ namespace Detail {
 
         class CoutStream final : public IStream {
             std::ostream m_os;
+            bool m_isatty;
         public:
             // Store the streambuf from cout up-front because
             // cout may get redirected when running tests
-            CoutStream() : m_os( Catch::cout().rdbuf() ) {}
+            CoutStream() : m_os( Catch::cout().rdbuf() ) {
+                m_isatty = true;
```

Comment:
Too much Copy&Paste for my taste - and since `Catch::cout()` can be reimplemented, this is quite possibly probing the wrong file handles...